integration/benchmark/main.go

```diff
@@ -0,0 +1,372 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	flowsdk "github.com/onflow/flow-go-sdk"
+	client "github.com/onflow/flow-go-sdk/access/grpc"
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/integration/utils"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type LoadCase struct {
+	tps      int
+	duration time.Duration
+}
+
+// This struct is used for uploading data to BigQuery, changes here should
+// remain in sync with tps_results_schema.json
+type dataSlice struct {
+	GitSha              string
+	StartTime           time.Time
+	EndTime             time.Time
+	InputTps            float64
+	OutputTps           float64
+	ProStartTransaction float64
+	ProEndTransaction   float64
+}
+
+func main() {
+	// holdover flags from loader/main.go
+	sleep := flag.Duration("sleep", 0, "duration to sleep before benchmarking starts")
+	loadTypeFlag := flag.String("load-type", "token-transfer", "type of loads (\"token-transfer\", \"add-keys\", \"computation-heavy\", \"event-heavy\", \"ledger-heavy\", \"const-exec\")")
+	chainIDStr := flag.String("chain", string(flowsdk.Emulator), "chain ID")
+	access := flag.String("access", net.JoinHostPort("127.0.0.1", "3569"), "access node address")
+	serviceAccountPrivateKeyHex := flag.String("servPrivHex", unittest.ServiceAccountPrivateKeyHex, "service account private key hex")
```

Comment:
these can be hardcoded for the CI test.

Reply:
This and several other flags have been moved to hardcoded variables as they will never change in CI testing.